p2p/protocol/identify/id.go

```diff
@@ -85,12 +89,14 @@ type IDService struct {
 
 	addrMu sync.Mutex
 
+	peerrec   *record.Envelope
+	peerrecMu sync.RWMutex
```

Comment:
Nit: mutex-hat (lock goes above the record)

Reply:
@Stebalien We will anyways get rid of this once we have the `GetRecord` function in the next PR.